framework/include/base/Adaptivity.h

```diff
@@ -300,6 +302,9 @@ class Adaptivity : public ConsoleStreamInterface,
   /// Name of the marker variable if using the new adaptivity system
   std::string _marker_variable_name;
 
+  /// Type of mesh adaptivity
+  std::string _adaptivity_type;
```

Comment:
```suggestion
  enum class AdaptivityType
  {
    H = 0,
    P = 1,
    HP = 2
  };

  /// Type of mesh adaptivity
  AdaptivityType _adaptivity_type;
```

Reply:
The enum class definition will need to be moved earlier in the class, before the `init` declaration

Reply:
You could also move it outside the `Adaptivity` class definition if you want to avoid constant `Adaptivity::` scoping